diff --git a/debian/configure b/debian/configure
index a0f5a8d..cc348bd 100755
--- a/debian/configure
+++ b/debian/configure
@@ -94,7 +94,7 @@ if [ ! -z "$RTAI_DEV" ]; then
 fi
 # libxenomai-dev does encode the architectures, but we have to resort to awful
 # shell juju to transform it into a proper dependency list
-XENOMAI_DEV="$(apt-cache search -n libxenomai-dev | head -1 | awk '{print $1}')"
+#XENOMAI_DEV="$(apt-cache search -n libxenomai-dev | head -1 | awk '{print $1}')"
 if [ ! -z "$XENOMAI_DEV" ]; then
     if [ `apt-cache showsrc libxenomai-dev 2>/dev/null | wc -l` -eq 0 ]; then
         cat 1>&2 <<-EOF
diff --git a/debian/rules.in b/debian/rules.in
index 566aa5e..fe921ae 100644
--- a/debian/rules.in
+++ b/debian/rules.in
@@ -33,7 +33,7 @@ build-stamp:
 # Add here commands to compile the package.
 	cd src && ./autogen.sh
 ifeq "$(kernel_version)" "uspace"
-	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=uspace --enable-build-documentation=pdf --disable-check-runtime-deps
+	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=uspace --enable-build-documentation=pdf --disable-check-runtime-deps XENOMAI_CONFIG=/usr/xenomai/bin/xeno-config
 else
 	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=/usr/realtime-$(kernel_version) --enable-build-documentation=pdf --disable-check-runtime-deps
 endif
@@ -47,7 +47,7 @@ clean: debian/control
 
 	cd src && ./autogen.sh
 ifeq "$(kernel_version)" "uspace"
-	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=uspace --enable-build-documentation=pdf --disable-check-runtime-deps
+	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=uspace --enable-build-documentation=pdf --disable-check-runtime-deps XENOMAI_CONFIG=/usr/xenomai/bin/xeno-config
 else
 	cd src && ./configure --prefix=/usr --sysconfdir=/etc --mandir=/usr/share/man --with-realtime=/usr/realtime-$(kernel_version) --enable-build-documentation=pdf --disable-check-runtime-deps
 endif
diff --git a/debian/shlibs.pre b/debian/shlibs.pre
index d55a559..d01a99b 100644
--- a/debian/shlibs.pre
+++ b/debian/shlibs.pre
@@ -1,2 +1,4 @@
 libncurses 5 libncurses5 (>= 5.0)
 libpth 20 libpth2 | libpth20
+libmodechk 0 libmodechk
+libcobalt 2 libcobalt
diff --git a/src/rtapi/uspace_rtapi_app.cc b/src/rtapi/uspace_rtapi_app.cc
index 7d503e2..48330c4 100644
--- a/src/rtapi/uspace_rtapi_app.cc
+++ b/src/rtapi/uspace_rtapi_app.cc
@@ -749,7 +749,7 @@ static int harden_rt()
     sig_act.sa_sigaction = signal_handler;
     sig_act.sa_flags   = SA_SIGINFO;
 
-    sigaction(SIGSEGV, &sig_act, (struct sigaction *) NULL);
+    //sigaction(SIGSEGV, &sig_act, (struct sigaction *) NULL);
     sigaction(SIGILL,  &sig_act, (struct sigaction *) NULL);
     sigaction(SIGFPE,  &sig_act, (struct sigaction *) NULL);
     sigaction(SIGTERM, &sig_act, (struct sigaction *) NULL);
diff --git a/src/rtapi/uspace_xenomai.cc b/src/rtapi/uspace_xenomai.cc
index 35151d6..580a351 100644
--- a/src/rtapi/uspace_xenomai.cc
+++ b/src/rtapi/uspace_xenomai.cc
@@ -1,7 +1,10 @@
 #include "config.h"
 #include "rtapi.h"
 #include "rtapi_uspace.hh"
-#include <posix/pthread.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <pthread.h>
 #include <atomic>
 #ifdef HAVE_SYS_IO_H
 #include <sys/io.h>
